app/src/test/java/com/codesoom/assignment/controllers/TaskControllerWebTest.java

```diff
@@ -0,0 +1,247 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.application.TaskService;
+import com.codesoom.assignment.models.Task;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.BDDMockito.given;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Nested
+@DisplayName("TaskControllerWebTest 클래스")
+@SpringBootTest
+@AutoConfigureMockMvc
+public class TaskControllerWebTest {
+
+    private final String[] TASK_TITLE = {"test1", "test2", "test3", "test4", "test5"};
+    private final String TASK_UPDATE = "update";
+    private final Long VALID_ID = 1L;
+    private final Long INVALID_ID = 100L;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @MockBean
+    private TaskService taskService; // 스프링에서 DI 주입
+
+
+    @Nested
+    @DisplayName("list 메소드는")
+    class Describe_list {
+
+        @Nested
+        @DisplayName("tasks 리스트에 값이 존재하면")
```

Comment:
컨트롤러에 알려져서는 안 되는 `tasks`가 리스트로 관리되고 있다는 사실이 유출되었습니다! 이건 `TaskService`의 비밀이에요.
```suggestion
        @DisplayName("등록된 할 일들이 1개 이상 있다면")
```

Reply:
수정하였습니다 !